File: 2022/day_01/haskell/app/Main.hs
module Main (main) where

import Data.List.Split (splitOn)
import Inventory

-- Groups input file lines by inventory.
groupLines :: [String] -> [[String]]
groupLines = splitOn [""]

main :: IO ()
main = do
    inputFile <- readFile "../input.txt"
    let topThreeInventories = getTopThreeInventories $ sumInventories $ linesToInventories $ groupLines $ lines inputFile
    let result1 = head topThreeInventories
    let result2 = sum topThreeInventories
    print ("Part 1 answer: " ++ show result1)
    print ("Part 2 answer: " ++ show result2)

File: 2022/day_01/rust/src/main.rs
use inventory::{lines_to_inventories, sum_first_three_calories, sum_inventories};
use std::fs;

mod inventory;

/// Reads problem input from file.
fn read_input_file() -> Vec<String> {
    let input_file_path = "../input.txt";
    let mut input_file_lines: Vec<String> = Vec::new();
    for line in fs::read_to_string(input_file_path).unwrap().lines() {
        input_file_lines.push(line.to_string());
    }
    input_file_lines
}

fn main() {
    let input_file_lines = read_input_file();
    let inventories = lines_to_inventories(input_file_lines);
    let inventory_sums = sum_inventories(inventories);
    let first_three_calories_sum = sum_first_three_calories(&inventory_sums);
    println!("Part 1 answer: {}", inventory_sums[0]);
    println!("Part 2 answer: {}", first_three_calories_sum);
}
